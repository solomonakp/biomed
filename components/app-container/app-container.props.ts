import { PropsWithChildren } from 'react';

type AppContainerBaseProps = PropsWithChildren<object>;

type HTMLMainElementAttributes = React.ComponentPropsWithoutRef<'main'>;

export type AppContainerProps = AppContainerBaseProps &
    HTMLMainElementAttributes;
