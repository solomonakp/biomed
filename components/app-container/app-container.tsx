interface AppContainerProps {
    children: React.ReactNode;
}

export const AppContainer = ({ children }: AppContainerProps) => {
    return (
        <main className="app-container w-full p-4 sm:p-6 md:p-8">
            <div className="mx-auto max-w-[1240px] p-6">{children}</div>
        </main>
    );
};
