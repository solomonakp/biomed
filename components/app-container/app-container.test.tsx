import { render, screen, cleanup } from '@testing-library/react';
import { axe } from 'jest-axe';
import { AppContainer } from './app-container';
import { AppContainerProps as Props } from './app-container.props';

const defaultProps: Props = {
    children: 'hi',
};

afterEach(cleanup);

const setUpAppContainer = (props: Props = defaultProps) =>
    render(<AppContainer {...props}>{defaultProps.children}</AppContainer>);

describe('App Container', () => {
    it('should the App Container component be exported correctly', () => {
        expect(AppContainer).toBeDefined();
    });
    test('should the App Container component be mounted correctly', () => {
        setUpAppContainer();

        const appContainer = screen.getByRole('main');
        expect(appContainer).toHaveTextContent(defaultProps.children as string);

        expect(appContainer).toBeInTheDocument();
    });

    test('should render the App Container user would see', () => {
        const { asFragment } = setUpAppContainer();

        expect(asFragment()).toMatchSnapshot();
    });
    test('should the App Container not fail any accessibility test', async () => {
        const { container } = setUpAppContainer();

        const result = await axe(container);

        expect(result).toHaveNoViolations();
    });
});
