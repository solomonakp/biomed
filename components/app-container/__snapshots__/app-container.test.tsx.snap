// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`App Container should render the App Container user would see 1`] = `
<DocumentFragment>
  <main
    class="app-container w-full p-4 sm:p-6 md:p-8"
  >
    <div
      class="mx-auto max-w-[1240px] p-6"
    >
      hi
    </div>
  </main>
</DocumentFragment>
`;
